test/files/pos/t4225.scala

```diff
@@ -0,0 +1,22 @@
+object Test {
+  class Foo {
+    class Bar
+    object Bar {
+      implicit def mkBar: Bar = new Bar
+    }
+  }
+
+  implicit class Ops[F <: Foo](val f0: F) {
```

Comment:
A test variant with a by-name implicit `f0` here might show up a problem with evaluation order by hoisting out the val. We deal with the same sort of issue in the named/default args transform, we would need to share some of that implementation in this case.